components/common/Cargo.toml

```diff
@@ -48,6 +48,13 @@ toml = { version = "*", features = ["preserve_order"] }
 uuid = { version = "*", features = ["v4"] }
 valico = "*"
 
+[target.'cfg(not(windows))'.dependencies]
+# Pulling from git until
+# https://github.com/nix-rust/nix/commit/3c2107bdc221a90b02d3f7118dd96f3496762cd0
+# is released, which will be the next release after v0.17.0
+# (A similar pin is in components/core and components/launcher as well).
+nix = {  git = "https://github.com/nix-rust/nix.git", rev = "3c2107bdc221a90b02d3f7118dd96f3496762cd0" }
```

Comment:
Not necessary to change here, but in the future, we could utilize [`[patch.crates-io]`](https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/replacing-dependencies-with-patch.html) to patch a dependency at the workspace level. Although I am not sure if it works with the `'cfg(not(windows))'` condition.

Reply:
Ah, good suggestion! I'll give that a try.

Reply:
Addressed in https://github.com/habitat-sh/habitat/pull/7823/commits/71c82566aad09d69bda334c5229992470c863410